tests/flow/test_graph_info.py

```diff
@@ -83,16 +83,34 @@ def consumeStream(self, stream, drop=True):
 
         self.env.assertEquals(t, "stream")
 
-        # consume entire stream
-        events = self.conn.xrevrange(stream, '+', '-')
-
         # convert stream events to LoggedQueries
-        logged_queries = [LoggedQuery(e[1]) for e in events]
+        logged_queries = []
+        streams = {stream: '0-0'}
+
+        elapsed = 10
+        while len(logged_queries) < n_items and elapsed > 0:
+            # read messages from the stream
+            messages = self.conn.xread(streams, block=0)
+
+            # process each message received
+            stream_messages = messages[0][1]
+            for message_id, message_payload in stream_messages:
+                # logged_queries.insert(0, LoggedQuery(message_payload))
+                logged_queries.append(LoggedQuery(message_payload))
+
+            if stream_messages:
+                # update stream last ID
+                streams[stream] = stream_messages[-1][0]
+
+            time.sleep(0.2)
+            elapsed -= 0.2
 
         # drop stream
         if drop:
             self.conn.delete(stream)
 
+        logged_queries.reverse()
```

Comment:
add a comment explaining why do we reverse

Reply:
I added a comment, assuming that there's a reason that you previously consumed the data using the `revrange` command, rather than the `range` command (~last in first out). Using `xread` we get the events by order, so we can just reverse the expected order of events as well..